integration/spark2/src/test/scala/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapSuite.scala

```diff
@@ -377,6 +377,102 @@ class BloomCoarseGrainDataMapSuite extends QueryTest with BeforeAndAfterAll with
     checkQuery("fakeDm", shouldHit = false)
   }
 
+  test("test create bloom datamap on newly added column") {
```

Comment:
Can this PR work for Lucene datamap? I think it should be a general fix for all index datamap, right?

Reply:
mostly yes, it depends on the way lucene deals with dictionary column. This PR enables to get default value of newly added column by a new result collector in `IndexDataMapRebuildRDD` and other codes are worked for getting surrogate value of dictionary column relying  on PR2425